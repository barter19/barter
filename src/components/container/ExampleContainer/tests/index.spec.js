import { shallow, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import ExampleContainerComponent from '../index';

const renderer = require('vue-server-renderer').createRenderer();

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ExampleContainer Component', () => {
  let store;
  let state;
  let actions;
  let getters;

  beforeEach(() => {
    state = {
      name: 'John Doe',
      count: 2,
    };

    actions = {
      updateName: jest.fn(),
      incrementCounter: jest.fn(),
      decrementCounter: jest.fn(),
    };

    getters = {
      positiveCount: () => 2,
      fibonacciNumber: () => 1,
    };

    store = new Vuex.Store({
      modules: {
        exampleContainer: {
          namespaced: true,
          state,
          actions,
          getters,
        },
      },
    });
  });

  it('dispatches "UPDATE_NAME" action when input is changed', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });
    const input = wrapper.find('input');

    // Simulate Add Text
    input.element.value = 'test';
    input.trigger('input');

    // Assert Result
    expect(actions.updateName.mock.calls).toHaveLength(1);

    // Simulate Remove Text
    input.element.value = '';
    input.trigger('input');

    // Assert Result
    expect(actions.updateName.mock.calls).toHaveLength(2);
  });

  it('dispatches "DECREMENT" action when button is clicked', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });

    // Simulate Click
    wrapper.find('button:first-of-type').trigger('click');

    // Assert Result
    expect(actions.decrementCounter.mock.calls).toHaveLength(1);
  });

  it('dispatches "INCREMENT" action when button is clicked', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });

    // Simulate Click
    wrapper.find('button:last-of-type').trigger('click');

    // Assert Result
    expect(actions.incrementCounter.mock.calls).toHaveLength(1);
  });

  it('renders values from store state', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });

    // Rendered Locations
    const h1Span = wrapper.find('h1 span');
    const h2Span = wrapper.find('h2 span');

    // Assert Result
    expect(h1Span.text()).toBe(state.name);
    expect(h2Span.text()).toBe(state.count.toString());
  });

  it('renders values from getters', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });

    // Rendered Locations
    const h3SpanFirst = wrapper.find('h3 span:first-of-type');
    const h3SpanLast = wrapper.find('h3 span:last-of-type');

    // Assert Results
    expect(h3SpanFirst.text()).toBe(getters.positiveCount().toString());
    expect(h3SpanLast.text()).toBe(getters.fibonacciNumber().toString());
  });

  it('renders to a nice snapshot', () => {
    const wrapper = shallow(ExampleContainerComponent, { store, localVue });
    const { vm } = wrapper;

    renderer.renderToString(vm, (err, str) => {
      expect(str).toMatchSnapshot();
    });
  });
});
