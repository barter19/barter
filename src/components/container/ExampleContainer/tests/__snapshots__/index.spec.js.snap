// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleContainer Component renders to a nice snapshot 1`] = `

<section data-server-rendered="true"
         class="example-container"
>
  <div class="container">
    <h1 class="mt-5">
      Name:
      <span>
        John Doe
      </span>
    </h1>
    <div class="input-group mb-3">
      <div class="input-group-prepend">
        <span id="update-name"
              class="input-group-text"
        >
          Update Name
        </span>
      </div>
      <input type="text"
             aria-describedby="update-name"
             placeholder="John Doe"
             class="form-control"
      >
    </div>
  </div>
  <div class="container">
    <h2 class="mt-5">
      Count:
      <span>
        2
      </span>
    </h2>
    <div role="group"
         aria-label="Basic example"
         class="btn-group"
    >
      <button type="button"
              class="btn btn-secondary"
      >
        Decrement
      </button>
      <button type="button"
              class="btn btn-secondary"
      >
        Increment
      </button>
    </div>
  </div>
  <div class="container">
    <h3 class="mt-5">
      Fibonacci Number
      <span>
        2
      </span>
      :
      <span>
        1
      </span>
    </h3>
  </div>
</section>

`;
